src/api.rs

```diff
@@ -506,6 +506,11 @@ impl Context {
       }
     }
 
+    match self.frame_q.get(&fi.number) {
```

Comment:
This could be something along the lines of
``` rust
if self.frame_q.get().map_or(false, |v| v.is_some()) { return Err(fi) }
```
It is obviously debatable what's more readable :)

Reply:
This caused the infinite loop to return. It could probably be fixed to have the same logic as the match statement, but given that it would require extra effort to do so, I think the `match` is clearer at showing the logic we want.